### PR TITLE
storage: add checksum logic in row slice, add cop and get test cases

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What is changed and how it works?

Issue Number: Ref #14528 

What's Changed:
Add checksum calculation logic in `RowSlice`, add `cop` and `kv-get` test cases, the target is to verify the checksum encoding impact on read paths.

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tidb_query_datatype/src/codec/row/v2/row_slice.rs [hunk 1]
```diff
@@ -10,42 +10,87 @@ use num_traits::PrimInt;
...
         let non_null_cnt = data.read_u16_le()? as usize;
         let null_cnt = data.read_u16_le()? as usize;
-        let row = if is_big {
+        let mut row = if is_big {
             RowSlice::Big {
                 origin,
                 non_null_ids: read_le_bytes(&mut data, non_null_cnt)?,
                 null_ids: read_le_bytes(&mut data, null_cnt)?,
                 offsets: read_le_bytes(&mut data, non_null_cnt)?,
                 values: LeBytes::new(data),
```
Comment (zyguan) [inline components/tidb_query_datatype/src/codec/row/v2/row_slice.rs:92]: `values` here includes checksum part, maybe cut the checksum suffix [like tidb does](https://github.com/pingcap/tidb/pull/42859/files#diff-204c31e3bed4805eefcfff1f62863268dfbedb8b81d066223fe4e58ef6527ddaR154) for safety.

File: components/tidb_query_datatype/src/codec/row/v2/encoder_for_test.rs [hunk 2]
```diff
@@ -217,6 +369,9 @@ pub trait RowEncoder: NumberEncoder {
         for col in non_null_cols {
             non_null_ids.push(col.id);
             value_wtr.write_value(ctx, &col)?;
+            if let Some(checksum_handler_ref) = checksum_handler.as_mut() {
+                checksum_handler_ref.checksum(&col)?;
+            }
```
[the remark this answers is not in the document]
Comment (cfzjywxk) [reply components/tidb_query_datatype/src/codec/row/v2/encoder_for_test.rs]: The checksum value is changed as a parameter.

File: components/tidb_query_datatype/src/codec/row/v2/row_slice.rs (hunk 1 above)
Comment (cfzjywxk) [reply components/tidb_query_datatype/src/codec/row/v2/row_slice.rs:92]: The cut logic is wrapper in the `cut_checksum_value` function, PTAL.

**Review by zyguan — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**